src/status_im/chat/models/loading.cljs

```diff
@@ -101,6 +101,12 @@
                                     :on-success #(re-frame/dispatch
                                                   [::mark-all-read-in-community-successful %])}]}))
 
+(defn mark-albums [messages]
```

Comment:
@OmarBasem, this function goes completely against the grain in Clojure. All you need is to map and assoc a new key. Stateful processing is rarely needed in Clojure.

```clojure
(defn mark-albums [messages]
  (map (fn [msg]
         (if (:album-id message)
           (assoc message :albumize true)
           message))
       messages))
```

Something even better IMO is to create smaller predicate functions and reuse them in collection operations. This leads to more functional code, that composes better. Of course, there are exceptions to this recommendation.

```clojure
(defn mark-album [message]
  (if (:album-id message)
    (assoc message :albumize true)
    message))

;; And you can use it as:
new-messages (map mark-album new-messages)
```

Reply:
why do we even need new flag? why not just use `album-id` as flag ?

Reply:
We a user receives a group of images while inside the chat, we do not group them. We group images into albums when a user is entering the chat screen. The `messages-loaded` function is called when a user is entering the chat screen, that's why I am adding the `albumize` boolean here.

Reply:
Thanks @ilmotta for the snippet

Reply:
@ilmotta do not why but both of these snippets did not work

Reply:
Although I didn't test the snippets, I'm curious enough now to have coded this in a REPL, and it works as expected @OmarBasem. But perhaps I'm missing something.

```clojure
(defn mark-albums
  [messages]
  (let [new-messages (atom [])]
    (doseq [message messages]
      (swap! new-messages conj (if (:album-id message) (assoc message :albumize true) message)))
    @new-messages))

(defn mark-album
  [message]
  (if (:album-id message)
    (assoc message :albumize true)
    message))

(let [messages [{:id "0x1" :album-id 1}
                {:id "0x2"}
                {:id "0x3" :album-id 3}]
      expected [{:id "0x1" :album-id 1 :albumize true}
                {:id "0x2"}
                {:id "0x3" :album-id 3 :albumize true}]]
  ;; Original
  (assert (= expected
             (mark-albums messages)))

  ;; New implementation
  (assert (= expected
             (map mark-album messages))))

```

Reply:
@ilmotta sorry my bad! It is working. I updated `albumize` to `albumize?` and did not change it in the snippet here. Thank you!

Reply:
Thanks for giving another try @OmarBasem o/